static/js/publisher/builds/components/searchSelect.js

```diff
@@ -0,0 +1,408 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class SearchSelect extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // The available values should be the full list minus the current values
+    // sorted by name
+    const values = this.props.values
+      .filter(value => !this.props.value.includes(value.key))
+      .sort(this.sortByName);
+
+    this.state = {
+      // Get the correct objects for the currentValues as selected
+      selected: this.props.values.filter(value =>
+        this.props.value.includes(value.key)
+      ),
+      values: values,
+      searchTerm: "",
+      searchResults: values,
+      showSearch: false,
+      highlightedOption: null,
+      isMulti: this.props.isMulti || false,
+      placeholder: this.props.placeholder || ""
+    };
+
+    // Bind all the things
+    // Clicking outside the component should hide the dropdown
+    this.blur = this.blur.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.removeItem = this.removeItem.bind(this);
+    this.clickItem = this.clickItem.bind(this);
+    this.addItem = this.addItem.bind(this);
+    this.focusInput = this.focusInput.bind(this);
+    this.search = this.search.bind(this);
+    this.handleKeypress = this.handleKeypress.bind(this);
+    this.clearAll = this.clearAll.bind(this);
+  }
+
+  sortByName(a, b) {
+    return a.name.localeCompare(b.name);
+  }
+
+  /**
+   * Remove an item from the 'selected' list.
+   * Add it back into the 'values' list.
+   */
+  removeItem(key) {
+    const toRemove = this.state.selected.filter(item => item.key === key)[0];
```

Comment:
I know this is probably my code, but we should use destructuring for getting state and props:
```
const {selected, values} = this.state;
const {getSelectedItems} = this.props;
```

Then in the function we can just have things like:
```
const newValues = values.slice(0);
const newSelected = selected.filter(...);

... getSelectedItems(selected);
```